### PR TITLE
[UCSC-88] Featured Media & Text

### PR DESCRIPTION
**Pull request by bjcooper**
Adds gradient, block pattern, and styles for a "Media with text, featured" block.

Ticket: https://moderntribe.atlassian.net/browse/UCSC-88

### CODE REVIEW

**Review by MlKilderkin — APPROVED** [review]

**Review by knice — APPROVED** [review]

**Comment by bjcooper** [comment]
Bump for CR @Vinsanity 🥂

File: src/scss/utilities/_mixins.scss [hunk 1]
```diff
@@ -1,6 +1,9 @@
 // Screen width breakpoints
 // Styles will take effect 1px above these sizes
 $small: 602px;
+// The breakpoint where WP Media & Text blcoks switch from being stacked for
+// mobile to their normal desktop layout.
+$wp-media-with-text-unstack: 601px;
```
Comment (Vinsanity) [inline src/scss/utilities/_mixins.scss:6]: Is there any reason the $small breakpoint won't work here? My concern is that we're adding some edge-case breakpoints that may only have one use.

Comment (bjcooper) [reply src/scss/utilities/_mixins.scss:6]: @Vinsanity , there are two reasons. **(1)** 601px is the actual screen size at which WP core unstacks their Media with Text (currently). If we use 602px, the layout is completely broken for viewports that are 601px wide. That's not likely a huge number of devices, but the cost to avoid it completely is to simply add an appropriate breakpoint size. **(2)** Less concretely, these layout changes are specifically tied to WP core unstacking their Media with Text—it's not related to any design/layout system of ours and the similarity to our `$small` breakpoint size is just a coincidence; they're not logically connected. I suppose WP core could tweak this number at any time in the future. In that case, we just have to update one SCSS var.

In short, this _is_ a very specific edge-case breakpoint that we may only use once (specifically, it's for if you need to apply a style to a Media & Text block that is unstacked). But the edge case exists and needs to be covered.

Comment (bjcooper) [reply src/scss/utilities/_mixins.scss:6]: I want to point out that I added another similar breakpoint `$wp-columns-unstack: 782px;` in a PR a couple of weeks ago, that does the exact same thing for WP core Columns blocks. That one is already merged.

Comment (Vinsanity) [reply src/scss/utilities/_mixins.scss:6]: Thanks @bjcooper. That makes sense and thanks for the explanation.

**Review by Vinsanity — APPROVED** [review]
🟢
